pkg/flow/internal/controller/loader.go

```diff
@@ -403,9 +415,13 @@ func (l *Loader) EvaluateDependencies(parentScope *vm.Scope, c *ComponentNode) {
 		}
 		return nil
 	})
+
+	if l.globals.OnExportsChange != nil {
+		l.globals.OnExportsChange(l.cache.CreateModuleExports())
+	}
```

Comment:
Can we do anything to detect whether a call to this is necessary to avoid stress on the parent component? 

If we can't do anything easily, a TODO comment to handle it later would be appreciated.

Reply:
I have a solution but not sure if I like it so will commit this PR and open another PR with the change.